protostar/cheatable_starknet/callable_hint_locals/declare_hint_local.py

```diff
@@ -29,34 +33,25 @@ class DeclareHintLocal(CallableHintLocal):
     def __init__(
         self,
         contracts_controller: ContractsController,
-        project_compiler: ProjectCompiler,
+        contract_path_resolver: ContractPathResolver,
     ):
-        self._contracts_controller = contracts_controller
-        self._project_compiler = project_compiler
+        self._contract_path_resolver = contracts_controller
+        self._project_compiler = contract_path_resolver
 
     @property
     def name(self) -> str:
         return "declare"
 
     def _build(self) -> Callable:
         def declare(contract: int) -> DeclaredContract:
-            contract_identifier = short_string_to_str(contract)
+            contract_name = short_string_to_str(contract)
 
-            try:
-                compiled_class, contract_class = _compile_contract(
-                    contract_identifier=contract_identifier,
-                )
-            except CompilationException as ex:
-                raise CheatcodeException(
-                    self, f"Compilation of {contract_identifier} failed"
-                ) from ex
-            except ContractNameNotFoundException as ex:
-                raise CheatcodeException(
-                    self, f"No contract found for the name {ex.contract_name}"
-                ) from ex
+            compiled_class, contract_class = _compile_contract(
+                contract_name=contract_name,
+            )
 
             declared_class: DeclaredSierraClass = asyncio.run(
-                self._contracts_controller.declare_sierra_contract(
+                self._contract_path_resolver.declare_sierra_contract(
```

Comment:
this doesn't seem right (why is `declare_sierra_contract` called on this object?)